apps/omg/test/omg/state/core_test.exs

```diff
@@ -897,6 +898,255 @@ defmodule OMG.State.CoreTest do
+      state =
+        state
+        |> do_deposit(alice, %{amount: 100, currency: @not_eth, blknum: 2})
+        |> Core.exec(
+          create_recovered([{1000, 0, 0, alice}, {2, 0, 0, alice}], [{alice, @eth, 5}, {alice, @not_eth, 100}]),
+          fees
+        )
+        |> success?()
+
+      # not_eth currency is transferred in full - no surplus exists
+      state
+      |> Core.exec(create_recovered_fee_tx(1000, state.fee_claimer_address, @not_eth, 1), fees)
+      |> fail?(:surplus_in_token_not_collected)
+    end
+
+    test "all fees paid from multi-input/output transactions are claimable",
+         %{alice: alice, state_empty: state, fees: fees} do
+      not_eth_1 = <<123::160>>
+      not_eth_2 = <<234::160>>
+
+      state =
+        state
+        |> do_deposit(alice, %{amount: 10, currency: @eth, blknum: 1})
+        |> do_deposit(alice, %{amount: 10, currency: @not_eth, blknum: 2})
+        |> do_deposit(alice, %{amount: 10, currency: not_eth_1, blknum: 3})
+        |> do_deposit(alice, %{amount: 10, currency: not_eth_2, blknum: 4})
+        |> Core.exec(
+          create_recovered(
+            [{1, 0, 0, alice}, {2, 0, 0, alice}, {3, 0, 0, alice}, {4, 0, 0, alice}],
+            [{alice, @eth, 8}, {alice, @not_eth, 7}, {alice, not_eth_1, 6}, {alice, not_eth_2, 5}]
+          ),
+          fees
+        )
+        |> success?()
+
+      can_claim_fees = [
+        create_recovered_fee_tx(1000, state.fee_claimer_address, @eth, 2),
+        create_recovered_fee_tx(1000, state.fee_claimer_address, @not_eth, 3),
+        create_recovered_fee_tx(1000, state.fee_claimer_address, not_eth_1, 4),
+        create_recovered_fee_tx(1000, state.fee_claimer_address, not_eth_2, 5)
+      ]
+
+      %Core{pending_txs: pending_txs} = Core.claim_fees(state)
+
+      # Is it too low? NOTE: pending_txs are reversed with payment tx last on list
+      assert can_claim_fees == pending_txs |> Enum.take(4) |> Enum.reverse()
+    end
+
+    test "surpluses adding up for same-token-fees paid in a block", %{alice: alice, state: state, fees: fees} do
+      state =
+        state
+        |> Core.exec(create_recovered([{1000, 0, 0, alice}], @eth, [{alice, 5}]), fees)
+        |> success?()
+
+      # we can claim sum of the surpluses from 2 txs (one in setup & one above)
+      collected = 3 + 2
+
+      state
+      |> Core.exec(create_recovered_fee_tx(1000, state.fee_claimer_address, @eth, collected), fees)
+      |> success?()
+
+      # but we have to claim the exact amount
+      state
+      |> Core.exec(create_recovered_fee_tx(1000, state.fee_claimer_address, @eth, collected + 1), fees)
+      |> fail?(:claimed_and_collected_amounts_mismatch)
+      |> Core.exec(create_recovered_fee_tx(1000, state.fee_claimer_address, @eth, collected - 1), fees)
+      |> fail?(:claimed_and_collected_amounts_mismatch)
+    end
+
+    # this test takes ~26 seconds on my machine
+    @tag skip: true
```

Comment:
on 2nd thought, this is not unreasonable speed. We were doing our perf tests on multiple cores, this is just one here, and thats 65K ecrecovers there, 26 seconds makes sense :(.

I don't know what would be a quick fix here. I don't like a skipped test, 26 seconds of `mix test` is bad too. Proper fix would be to parametrize block size in `Core.extract...` saving 2**16 as a default for convenience, probably better put outside of the scope of this PR.

So my propo is this:
 - unskip
 - tag `:slow` so that `mix test --exclude slow` is possible for development, for the time being
 - (optionally, unless you just do it in a follow-up PR) put an issue to parametrize block size
 - describe in a NOTE here, or even better `Logger.warn("slow test is running, use --exclude slow to skip")`
 - don't forget about it :)